providers/unleash/src/main/java/dev/openfeature/contrib/providers/unleash/UnleashProvider.java

```diff
@@ -0,0 +1,171 @@
+            log.info("ready state will be changed via UnleashSubscriberWrapper events");
+        }
+
+        log.info("finished initializing provider, state: {}", state);
+    }
+
+    @Override
+    public Metadata getMetadata() {
+        return () -> NAME;
+    }
+
+    @Override
+    public void emitProviderReady(ProviderEventDetails details) {
+        super.emitProviderReady(details);
+        state = ProviderState.READY;
+    }
+
+    @Override
+    public void emitProviderError(ProviderEventDetails details) {
+        super.emitProviderError(details);
+        state = ProviderState.ERROR;
+    }
+
+    @Override
+    public ProviderEvaluation<Boolean> getBooleanEvaluation(String key, Boolean defaultValue, EvaluationContext ctx) {
+        if (!ProviderState.READY.equals(state)) {
+            if (ProviderState.NOT_READY.equals(state)) {
+                throw new ProviderNotReadyError(PROVIDER_NOT_YET_INITIALIZED);
+            }
+            throw new GeneralError(UNKNOWN_ERROR);
+        }
+        UnleashContext context = ctx == null ? UnleashContext.builder().build() : ContextTransformer.transform(ctx);
+        boolean featureBooleanValue = unleash.isEnabled(key, context, defaultValue);
+        return ProviderEvaluation.<Boolean>builder()
+            .value(featureBooleanValue)
+            .build();
+    }
+
+    @Override
+    public ProviderEvaluation<String> getStringEvaluation(String key, String defaultValue, EvaluationContext ctx) {
+        if (!ProviderState.READY.equals(state)) {
+            if (ProviderState.NOT_READY.equals(state)) {
+                throw new ProviderNotReadyError(PROVIDER_NOT_YET_INITIALIZED);
+            }
+            throw new GeneralError(UNKNOWN_ERROR);
+        }
+        UnleashContext context = ctx == null ? UnleashContext.builder().build() : ContextTransformer.transform(ctx);
+        Variant evaluatedVariant = unleash.getVariant(key, context);
+        String variantName;
+        String value;
+        if (DISABLED_VARIANT.equals(evaluatedVariant)) {
+            variantName = null;
+            value = defaultValue;
+        } else {
+            variantName = evaluatedVariant.getName();
+            value = evaluatedVariant.getPayload().get().getValue();
+        }
+        return ProviderEvaluation.<String>builder()
+            .value(value)
+            .variant(variantName)
+            .build();
+    }
+
+    @Override
+    public ProviderEvaluation<Integer> getIntegerEvaluation(String key, Integer defaultValue, EvaluationContext ctx) {
+        throw new TypeMismatchError(NOT_IMPLEMENTED);
+    }
+
+    @Override
+    public ProviderEvaluation<Double> getDoubleEvaluation(String key, Double defaultValue, EvaluationContext ctx) {
+        throw new TypeMismatchError(NOT_IMPLEMENTED);
+    }
+
+    @Override
+    public ProviderEvaluation<Value> getObjectEvaluation(String s, Value value, EvaluationContext evaluationContext) {
+        throw new TypeMismatchError(NOT_IMPLEMENTED);
```

Comment:
I do not understand why we are not returning the variant payload here? It is quite complex and support a lot of things (json, CSV, string, numbers, etc).

Reply:
I think we probably should just return the entire variant object here.

Reply:
The `Value` type supports almost anything - it can return an array like structure or an object like structure, or even boxed basic types.

I agree that this can be enhanced to return CSV and JSON payloads from unleash as a Value containing Arrays and Maps, respectively.

Reply:
I think we should return the entire `Variant` object from Unleash as it contain more details than just the payload.

Reply:
I added some details to evaluation via [flag metadata](https://github.com/open-feature/java-sdk-contrib/compare/main...liran2000:java-sdk-contrib:feature/unleash?expand=1#diff-33b37a3e9b9bd14c4d0853c582e47b468d3927d41da53b47900a8c60a90b1133R47):

>Additional evaluation data can be received via flag metadata, such as:
  * *enabled* - boolean
  * *variant-stickiness* - string
  * *payload-type* - string, optional